### PR TITLE
Remove s2i extended build documentation

### PR DESCRIPTION
**Pull request by csrwng**
Removes extended build documentation from the docs because we are deprecating the feature and adds documentation on how this can be achieved using image source and chaining builds.

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees ptal 
I wasn't sure where to best put this doc, so I put it in the advanced build operations section. Let me know if you think it can go in a better spot.

File: dev_guide/builds/advanced_build_operations.adoc [hunk 1]
```diff
@@ -122,3 +122,123 @@ endif::[]
 If the specified `NodeSelector` cannot be matched to a node with those labels,
 the build still stay in the `Pending` state indefinitely.
 ====
+
+[[dev-guide-chaining-builds]]
+== Chaining Builds
+
+For compiled languages (Go, C, C++, Java, etc.) the dependencies necessary for compilation might increase the size of the
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: s/the dependencies necessary for compilation/including the dependencies necessary for compilation in the application image/

File: dev_guide/builds/advanced_build_operations.adoc [hunk 2]
```diff
@@ -122,3 +122,123 @@ endif::[]
...
+image or introduce vulnerabilities that can be exploited.
+
+To avoid these problems, a
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#source-build[Source-to-Image]
+build can be chained with a
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#docker-build[Docker]
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: any 2 build types can be chained together, it doesn't have to be s2i -> docker.

File: dev_guide/builds/advanced_build_operations.adoc [hunk 3]
```diff
@@ -122,3 +122,123 @@ endif::[]
...
+                                       |
+                                       v
+                               +---------------+
+                               | Runtime Image |
+                               | with Compiled |
+                               |    Artifact   |
+                               +---------------+
+----
+
+The first build, which takes source and produces a binary artifact, is a a regular Source-to-Image
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: s/a a/a/

File: dev_guide/builds/advanced_build_operations.adoc [hunk 4]
```diff
@@ -122,3 +122,123 @@ endif::[]
...
+build. In the following example, a JEE source repository is to build a WAR file that is output to
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: "a JEE source repository is used to build an image containing a WAR file.  The image is output to(pushed to?) the `artifact-image` ImageStream."

File: dev_guide/builds/advanced_build_operations.adoc [hunk 5]
```diff
@@ -122,3 +122,123 @@ endif::[]
...
+metadata:
+  name: artifact-build
+spec:
+  output:
+    to:
+      kind: ImageStreamTag
+      name: artifact-image
+  source:
+    git:
+      uri: https://github.com/bparees/openshift-jee-sample.git
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: i think it's time to put a copy of this repo in the openshift org, i don't need my name in the official product docs :)

Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: i've moved it into openshift.

File: dev_guide/builds/advanced_build_operations.adoc [hunk 6]
```diff
@@ -122,3 +122,123 @@ endif::[]
...
+    sourceStrategy:
+      from:
+        kind: ImageStreamTag
+        name: wildfly:10.1
+        namespace: openshift
+    type: Source
+----
+
+
+The second build uses Image Source and, in this case, an inline Dockerfile to place the binary artifact resulting
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: link to the imagesource doc

**Review by bparees — CHANGES_REQUESTED** [review]
advanced operations seems like as a good a place as any.

File: dev_guide/builds/advanced_build_operations.adoc (hunk 2 above)
Comment (csrwng) [inline dev_guide/builds/advanced_build_operations.adoc]: Yes they can, but this is about a specific solution to the problem of capturing binary output from one build and including it in a runtime image. Maybe we should have a different title?

**Comment by csrwng** [comment]
comments addressed

**Review by bparees — CHANGES_REQUESTED** [review]
needs a little more detail around the fact that the runtime image doesn't have/need any compilation tools in it, and pointing out that anytime the first build produces a new image, the second build will automatically be triggered due to the imagechangetrigger (but this is optional), making it effectively act as a single build flow.

**Comment by csrwng** [comment]
@bparees added a paragraph explaining what you said :)

File: dev_guide/builds/advanced_build_operations.adoc [hunk 7]
```diff
@@ -122,3 +122,132 @@ endif::[]
...
+might increase the size of the image or introduce vulnerabilities that can be exploited.
+
+To avoid these problems, two builds can be chained together. One that will produce the compiled artifact, and a second build
+that will place that artifact in a separate image that will run the artifact.
+
+In the following example, a
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#source-build[Source-to-Image]
+build is combined with a
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#docker-build[Docker]
+build to compile an artifact that is then placed in a separate runtime image.
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: "Note: although this example chains an s2i build and a docker build, the first build can use any strategy that will produce an image containing the desired artifacts, and the second build can use any strategy that can consume input content from an image."

(wordsmith as desired).

**Review by bparees — CHANGES_REQUESTED** [review]

File: dev_guide/builds/advanced_build_operations.adoc (hunk 7 above)
Comment (csrwng) [inline dev_guide/builds/advanced_build_operations.adoc]: done

**Review by bparees — APPROVED** [review]

**Comment by bparees** [comment]
lgtm @openshift/team-documentation ptal

**Comment by adellape** [comment]
@csrwng Picking this up via https://github.com/openshift/openshift-docs/pull/3829.

**Pull request closed without merging**